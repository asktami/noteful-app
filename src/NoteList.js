import React from 'react';

import NotefulContext from './NotefulContext';

import NoteItem from './NoteItem';

class NoteList extends React.Component {
	// const contextType is needed to then say this.context
	static contextType = NotefulContext;

	render() {
		const { notes } = this.context;

		// if selected a folder, show only the notes in that folder
		// otherwise show all notes from all folders
		const foldernotes = this.props.match.params.folderId
			? notes.filter(note => note.folderId == this.props.match.params.folderId)
			: notes;

		return (
			<>
				<div className="header-container">
					<span>
						<h2>Notes</h2>
					</span>
					&nbsp;&nbsp;
					<span>
						<button className="btn-add" onClick={this.props.togglePopup}>
							+
						</button>
					</span>
				</div>

				{foldernotes.map(note => (
					<article key={note.id}>
						<div className="note">
							<NoteItem {...note} />
						</div>
					</article>
				))}
			</>
		);
	}
}

export default NoteList;
