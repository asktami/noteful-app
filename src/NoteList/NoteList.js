import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import NoteError from './NoteError';

import NotefulContext from '../NotefulContext';

import NoteItem from '../NoteItem/NoteItem';

const NoteList = props => {
	// need to grab NotefulContext (globals)
	const contextType = useContext(NotefulContext);
	const { notes } = contextType;

	// if selected a folder, show only the notes in that folder
	// otherwise show all notes from all folders
	const foldernotes = props.match.params.folderId
		? notes.filter(note => note.folderId === props.match.params.folderId)
		: notes;

	return (
		<>
			<header>
				<h2>Notes</h2>
				&nbsp;&nbsp;
				<NavLink
					// if just passing url:
					// to={'/add-note'}

					// to pass selected folderId:
					to={{
						pathname: '/add-note',
						state: { folderId: props.match.params.folderId }
					}}
				>
					<button className="btn-add">+</button>
				</NavLink>
			</header>
			{foldernotes.length > 0 ? (
				foldernotes.map(note => (
					<section key={note.id}>
						<NoteError>
							<div className="note">
								{/*
							{...note}
							use spread here IF want whole note object, and then get inside NoteItem via note.key
							alternative is
							note={note} and in NoteItem get via props.note.key

							NOTE:
							MUST pass {...props} to have the history, location, and match props inside NoteItem
							*/}
								<NoteItem note={note} {...props} />
							</div>
						</NoteError>
					</section>
				))
			) : (
				<section className="no-border">
					<div className="note">No notes in this folder.</div>
				</section>
			)}
		</>
	);
};

export default NoteList;

// to catch bugs
// check that get a notes array that has id, name, and modified
// this array is the "notes" variable coming from context
NoteList.propTypes = {
	notes: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.string.isRequired,
			name: PropTypes.string.isRequired,
			modified: PropTypes.instanceOf(Date).isRequired
		})
	)
};
