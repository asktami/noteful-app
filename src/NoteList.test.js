import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import NoteList from './NoteList';

describe(`NoteList component`, () => {
	const props = {
		match: {
			params: {
				folderId: 101
			}
		},
		notes: [
			{
				id: 1,
				folderId: 101,
				title: 'test-note-title',
				modified: new Date(2019, 8, 30)
			}
		]
	};

	it('renders a NoteList by default', () => {
		const wrapper = shallow(<NoteList {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('renders the NoteList given props', () => {
		const wrapper = shallow(<NoteList {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
