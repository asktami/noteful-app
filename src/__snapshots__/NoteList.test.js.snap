// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList component renders a NoteList by default 1`] = `
<Fragment>
  <div
    className="header-container"
  >
    <span>
      <h2>
        Notes
      </h2>
    </span>
      
    <span>
      <button
        className="btn-add"
      >
        +
      </button>
    </span>
  </div>
  <article>
    <ul>
      <li
        key="1"
      >
        <NoteItem
          match={
            Object {
              "params": Object {
                "folderId": 101,
              },
            }
          }
          note={
            Object {
              "folderId": 101,
              "id": 1,
              "modified": 2019-09-30T04:00:00.000Z,
              "title": "test-note-title",
            }
          }
          notes={
            Array [
              Object {
                "folderId": 101,
                "id": 1,
                "modified": 2019-09-30T04:00:00.000Z,
                "title": "test-note-title",
              },
            ]
          }
        />
      </li>
    </ul>
  </article>
</Fragment>
`;

exports[`NoteList component renders the NoteList given props 1`] = `
<Fragment>
  <div
    className="header-container"
  >
    <span>
      <h2>
        Notes
      </h2>
    </span>
      
    <span>
      <button
        className="btn-add"
      >
        +
      </button>
    </span>
  </div>
  <article>
    <ul>
      <li
        key="1"
      >
        <NoteItem
          match={
            Object {
              "params": Object {
                "folderId": 101,
              },
            }
          }
          note={
            Object {
              "folderId": 101,
              "id": 1,
              "modified": 2019-09-30T04:00:00.000Z,
              "title": "test-note-title",
            }
          }
          notes={
            Array [
              Object {
                "folderId": 101,
                "id": 1,
                "modified": 2019-09-30T04:00:00.000Z,
                "title": "test-note-title",
              },
            ]
          }
        />
      </li>
    </ul>
  </article>
</Fragment>
`;
